source/Command.cpp

```diff
@@ -355,15 +355,21 @@ Command::operator bool() const
 // Check whether this command is entirely empty.
 bool Command::operator!() const
 {
-	return !state && !turn;
+	return state.none() && !turn;
 }
 
 
 
 // For sorting commands (e.g. so a command can be the key in a map):
 bool Command::operator<(const Command &command) const
 {
-	return (state < command.state);
+	static const int count = static_cast<int>(COMMAND_COUNT);
+	for(int i = count - 1; i >= 0; --i)
+		if(state[i] > command.state[i])
```

Comment:
You should use `.test(i)` here, since you only need to access the value and don't need to modify the bit.

Reply:
done